library/manageiq_user.py

```diff
@@ -83,180 +101,140 @@
     miq_verify_ssl: False
 '''
 
-import os
-from manageiq_client.api import ManageIQClient as MiqApi
+import manageiq_utils
 
 
 class ManageIQUser(object):
-    """ ManageIQ object to execute user management operations in manageiq
-
-    url            - manageiq environment url
-    user           - the username in manageiq
-    password       - the user password in manageiq
-    miq_verify_ssl - whether SSL certificates should be verified for HTTPS requests
-    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+        object to execute user management operations in manageiq
     """
 
-    def __init__(self, module, url, user, password, miq_verify_ssl, ca_bundle_path):
-        self.module        = module
-        self.api_url       = url + '/api'
-        self.user          = user
-        self.password      = password
-        self.client        = MiqApi(self.api_url, (self.user, self.password), verify_ssl=miq_verify_ssl, ca_bundle_path=ca_bundle_path)
-        self.changed       = False
-
-    def find_group_by_name(self, group_name):
-        """ Searches the group name in ManageIQ.
-
-        Returns:
-            the group id if it exists in manageiq, None otherwise.
-        """
-        groups = self.client.collections.groups
-        return next((group.id for group in groups if group.description == group_name), None)
-
-    def find_user_by_userid(self, userid):
-        """ Searches the userid in ManageIQ.
-
-        Returns:
-            the user's id if it exists in manageiq, None otherwise.
-        """
-        users = self.client.collections.users
-        return next((user.id for user in users if user.userid == userid), None)
+    def __init__(self):
+        self.changed = False
 
-    def delete_user(self, userid):
+    def delete_user(self, manageiq, userid):
         """Deletes the user from manageiq.
 
         Returns:
             a short message describing the operation executed.
         """
-        user_id = self.find_user_by_userid(userid)
-        if not user_id:  # user doesn't exist
+        user = manageiq.find_collection_resource_by('users', userid=userid)
+        if not user:  # user doesn't exist
             return dict(
                 changed=self.changed,
                 msg="User {userid} does not exist in manageiq".format(userid=userid))
         try:
-            url = '{api_url}/users/{user_id}'.format(api_url=self.api_url, user_id=user_id)
-            result = self.client.post(url, action='delete')
+            url = '{api_url}/users/{user_id}'.format(api_url=manageiq.api_url, user_id=user['id'])
+            result = manageiq.client.post(url, action='delete')
```

Comment:
I agree, but I prefer to add it in a separate PR, this one is quite a big change already.